File: frontend/webschedule-app/src/services/activity/dtos/activityResponse.ts
import { BuildingResponse } from "../../building/dtos/buildingResponse";
import { SessionResponse } from "../../session/dtos/sessionResponse";

export interface ActivityResponse {
    activityId: number;
    name: string;
    teacherFullName: string;
    room: string;
    startingHour: number;
    duration: number;
    weekDay: string;
    session: SessionResponse;
    building: BuildingResponse;
}

File: backend/WebSchedule/Responses/ActivityResponse.cs
namespace WebSchedule.Controllers.Responses
{
    public class ActivityResponse
    {
        public int ActivityId { get; set; }
        public string Name { get; set; }
        public string TeacherFullName { get; set; }
        public string Room { get; set; }
        public int StartingHour { get; set; }
        public int Duration { get; set; }
        public string WeekDay { get; set; }
        public SessionResponse Session { get; set; }
        public BuildingResponse Building { get; set; }
    }
}
